sage.py

```diff
@@ -1629,126 +1425,111 @@ def make_AG(condensed_v_data, condensed_data, state_groups, sev_sinks, datafile,
-    #range_ = str(sys.argv[5])
     try:
-        #range_ = range_.replace('(', '').replace(')', '').split(',')
-        _s_ = float(sys.argv[5])#int(range_[0])
-        _e_ = float(sys.argv[6])#int(range_[1])
-        print('Filtering alerts. Only parsing from %d-th to %d-th hour (relative to the start of the alert capture)'%(_s_,_e_))
-    except:
+        start_hour = float(sys.argv[5])
+        end_hour = float(sys.argv[6])
+        print('Filtering alerts. Only parsing from %d-th to %d-th hour (relative to the start of the alert capture)' % (start_hour, end_hour))
+    except (ValueError, TypeError) as e:
         print('Error parsing hour filter range')
         sys.exit()
 
-startTimes = [] # We cheat a bit: In case user filters to only see alerts from (s,e) range, 
-#                    we record the first alert just to get the real time-elapsed since first alert
-    
-outaddress = ""
+# We cheat a bit: In case user filters to only see alerts from (s,e) range,
+#   we record the first alert just to get the real time-elapsed since first alert
+start_times = []
+
 path_to_ini = "FlexFringe/ini/spdfa-config.ini"
 
-modelname = expname+'.txt'#'test-trace-uni-serGroup.txt'
-datafile = expname+'.txt'#'trace-uni-serGroup.txt'
-   
-path_to_traces = datafile
-
-print('------ Downloading the IANA port-service mapping ---------')
-port_services = load_IANA_mapping()
-
-print('----- Reading alerts ----------')
-(unparse, team_labels) = load_data(folder, t) # t = minimal window for alert filtering
-plt = plot_histogram(unparse, team_labels)
-plt.savefig('data_histogram-'+expname+'.png')
-print('------ Converting to episodes ---------')
-team_episodes,_ = aggregate_into_episodes(unparse, team_labels, step=w) # step = w
-print('\n---- Converting to episode sequences -----------')
-host_data  =  host_episode_sequences(team_episodes)
-print('----- breaking into sub-sequences and making traces----------')
-(alerts, keys) = break_into_subbehaviors(host_data)
-generate_traces(alerts, keys, datafile)
-
-
-print('------ Learning SPDFA ---------')
-# Learn S-PDFA
+path_to_traces = experiment_name + '.txt'
+
+print('------ Downloading the IANA port-service mapping ------')
+port_services = load_iana_mapping()
+
+print('------ Reading alerts ------')
+(team_alerts, team_labels) = load_data(path_to_json_files, alert_filtering_window)
+plot_histogram(team_alerts, team_labels)
+
+print('------ Converting to episodes ------')
+team_episodes, _ = aggregate_into_episodes(team_alerts, step=alert_aggr_window)
+
+print('\n------ Converting to episode sequences ------')
+host_data = host_episode_sequences(team_episodes)
+
+print('------ Breaking into sub-sequences and generating traces ------')
+episode_subsequences = break_into_subbehaviors(host_data)
+generate_traces(episode_subsequences, path_to_traces)
+
+
+print('------ Learning S-PDFA ---------')
 flexfringe(path_to_traces, ini=path_to_ini, symbol_count="2", state_count="4")
 
-## Copying files
```

Comment:
In this part, there were multiple variables referring to the same thing (`path_to_model`, `path_to_traces`, `datafile`, `modelname`, `outfile`). I have kept only `path_to_traces` to avoid the confusion